packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/repositories/counter_repository.dart

```diff
@@ -5,31 +5,19 @@
 // license that can be found in the LICENSE file or at
 // https://opensource.org/licenses/MIT.
 
-/// This will simulate a server with 100 milliseconds response time
+import '../../base/data_sources/domain_data_sources/count_data_sources/count_data_source.dart';
+import '../../base/models/count.dart';
+
+///Can use any type of data source, that follow the contract
+///and retrieve its response to the business logic layer
 class CounterRepository {
-  int _counter = 0;
 
-  /// Increment the stored counter by one
-  Future<int> increment() async {
-    // Server response time.
-    await Future.delayed(const Duration(milliseconds: 800));
-    // Simulate an error from the server when the counter reached 2.
-    if (_counter == 2) {
-      throw Exception('Maximum number is reached!');
-    }
+  CounterRepository(this.countDataSource);
 
-    return ++_counter;
-  }
+  final CountDataSource countDataSource;
 
-  /// Decrement the stored counter by one
-  Future<int> decrement() async {
-    // Server response time.
-    await Future.delayed(const Duration(milliseconds: 800));
-    // Simulate an error from the server when the counter reached 2.
-    if (_counter <= 0) {
-      throw Exception('Minimum number is reached!');
-    }
+  Future<Count> getCurrent()=> countDataSource.getCurrent();
```

Comment:
@RomanovaPrime, please add comments to the methods

Reply:
Added.